app/Http/Controllers/GithubController.php

```diff
@@ -40,4 +45,19 @@ public function callback(): RedirectResponse
 
         return redirect("/");
     }
+
+    public function fetchData($organizationId): RedirectResponse
+    {
+        $userId = Auth::user()->id;
+
+        FetchDataFromApi::dispatch(
+            (int)$organizationId,
+            $githubConnector = new GithubConnector(),
+            new FetchRepositoriesService($githubConnector, $userId),
+            new FetchWorkflowRunsService($githubConnector, $userId),
+            new FetchWorkflowJobsService($githubConnector, $userId),
+        );
+
+        return redirect()->back();
+    }
```

Comment:
Maybe for fetching data nice to be pack that in [queue](https://laravel.com/docs/11.x/queues)?